### ISSUE TITLE
ResAutoencoder forward pass inconsistent with Architecture diagram.

### ISSUE THREAD

**Issue by Ayushutkarsh** [Clench/TAR_resAE#1]
Respected Authors,

We tried training your `current` implementation on FaceSwap from FaceForensics++ dataset and noticed that the `act_loss` doesn't converge. It remains constant at around `1024`. Using the attached testing code, we got an accuracy of `49.5%` after 50 epochs. Is there anything we are missing?

Secondly, while probing for any misses from our side, we noticed that according to the architecture diagram in the paper (fig.2 ), input to the DECODER should be the split-latent activations after applying the facilitator latent-space updation on the ENCODER's output.
Consequently, In the implementation  provided in the attached notebook,  for class `ResAutoEncoder`  should the input to `self.decoder` not be `dep` ? i.e.  ` x = self.decoder(dep)`
`dep` is not used elsewhere in the code otherwise. 

```
def forward(self, x, label):        
        x = self.encoder(x)
        act = x.clone()
        dep = x.clone()
        # Selection block setting zero values based on label
        # [:64] -> fake data latent space
        # [64:] -> real data latent space
        # 0->fake, 1->real
        A = nn.Parameter(torch.zeros(64,15,15))
        
        for i in range(len(label)):
            # real
            if label[i].item():
                # setting fake latent space into zero
                dep[i, :64] = A
            else:
                dep[i, 64:] = A
                
        x = self.decoder(x)
        
        return x, act
```

**Comment by Blosslzy** [comment]
I also encountered this problem. Have you solved it?
